### ISSUE TITLE
`UseDataClass` false positive: "class defines no functionality"

### ISSUE THREAD

**Issue by yogurtearl** [detekt/detekt#6168]
Consider this code:

```kotlin
class Bar(
    val param1: String,
    foo: Int,
) {
    val baz = 2 * foo
}
```

## Expected Behavior
`UseDataClass` does not say anything. 

## Observed Behavior
detekt gives this warning with `UseDataClass` is enabled.
```
The class Bar defines no functionality and only holds data. Consider converting it to a data class. [UseDataClass]
```

"defines no functionality" is not true (see `val baz` )

if I follow the advice and add `data` before `class` I will get a compiler error because `foo: Int` is not a `val`: 
```
Data class primary constructor must only have property (val / var) parameters
```

## Context

Following detekt's advice leads to a compile error.

## Your Environment

* Version of detekt used: 1.23.0
* Version of Gradle used (if applicable): 8.1.1
